blog-content/5-make-source-maps-public.md

```diff
@@ -0,0 +1,81 @@
+---
+title: Your Source Maps Should Be Public
+createdAt: 2023-05-12T12:00:00Z
+readingTime: 8
+authorFirstName: Chris
+authorLastName: Esplin
+authorTitle: Software Engineer @ Highlight 
+authorTwitter: 'https://twitter.com/chrisesplin'
+authorLinkedIn: 'https://www.linkedin.com/in/epsilon/'
+authorGithub: 'https://github.com/deltaepsilon'
+authorWebsite: 'https://www.chrisesplin.com/'
+authorPFP: 'https://firebasestorage.googleapis.com/v0/b/quiver-pixels-2020.appspot.com/o/F1EQ3eaBqkbEKEHBigolXIlmdut2%2F1408a808-60a6-4102-b636-08ab24041503.jpeg?alt=media&token=5f0ed5d8-c192-4aa3-a75b-3eb6cac9a552'
+tags: Developer Tooling
+metaTitle: Your Source Maps Should Be Public
+---
+
+Source maps are critical for web development in today's JavaScript environment.
+
+All of our build tools—Rollup, Vite, WebPack, ESBuild—transpile and bundle our JavaScript and CSS.
+
+Debugging errors in production with transpiled code is nigh impossible. So our tooling creates source maps.
+
+See [What are source maps? by Jecelyn Yeen](https://web.dev/source-maps/) for all of the details.
+
+# Why private source maps?
+
+Private source maps make it harder for an attacker to understand your front-end application.
+
+An attacker can easily use source maps to view the code in its original, un-transpiled state. The same source maps that make debugging easier make attacking your application easier.
+
+Check out our unobfuscated source code on `https://www.highlight.io`!
```

Comment:
```
Check out our unobfuscated source code on `https://www.highlight.io`!
```

Maybe we can say something about how this is an example of a sourcemap-applied js bundle?

Reply:
??

```
Check out our unobfuscated source code on `https://www.highlight.io`! It's readable, public and easy to ingest for developer tools like Highlight. Our `vite` bundler generates these source maps with a [single flag](https://vitejs.dev/config/build-options.html#build-sourcemap).
```